R/tag_query.R

```diff
@@ -339,6 +339,21 @@ assertNotTagEnvLike <- function(x, fnName) {
 }
 
 
+# Shim in a class so that the print method of tagList() is not used
+# Yet knit print methods will work as if they are tagList objects.
+visibleTagList <- function(...) {
+  y <- tagList(...)
+  oldClass(y) <- c("htmltools.tag.list.visible", oldClass(y))
```

Comment:
I think that this should be `shiny.tag.list.visible`, to be consistent with the other class names we use. If we could go back in time and change them all to `htmltools.tag.*`, that would be nice, but we can't.

Reply:
Also, `"visible"` doesn't feel like quite the right name, and I feel that it's a little weird to add a new class just so that it'll print differently. How about just adding an attribute to the tagList, like `print_list=TRUE`, and then checking for that attribute in `print.shiny.tag.list`? (Note that `print_list` isn't a great name.)

Reply:
Rename: ✅ 

"Visible": I'm going to have `$selected()` (which is where `visibleTagList()` was called) return a `list()` and not a `tagList()`. This will solve our complications. (`$root()` will still return a `tagList` and not a `list()`)

Reply:
😞 I now remember why `visibleTagList()` was used. Going to use a `print.as.list = TRUE` attribute

Reply:
Just curious: why doesn't it work to return a `list()`?

Reply:
Rename: bd66aa4
`print.as.list` attr: 4f6cc0e